test/unit/proj_errno_string_test.cpp

```diff
@@ -0,0 +1,57 @@
+/******************************************************************************
+ *
+ * Project:  PROJ
+ * Purpose:  Unit test for proj_errno_string()
+ * Author:   Kurt Schwehr <schwehr@google.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, Kristian Evers.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "proj.h"
+
+#include "gtest_include.h"
+
+namespace {
+
+TEST(ProjErrnoStringTest, Basic) {
+
+    // Test the no error result
+    ASSERT_EQ(0, proj_errno_string(0));
+
+    // Test both valid and invalid PROJ error numbers
+    ASSERT_STREQ("no arguments in initialization list", proj_errno_string(-1));
```

Comment:
ASSERT_STREQ -> EXPECT_EQ.  Try to reserve asserts for things were it is totally unsafe to proceed.  By using expect, you get to keep generating test output and fail later.  That often makes debugging easier when I do something dumb to the code under test.

Reply:
I just did a search for how to test if two strings are equal and this came up. I'll change the ASSERTs to EXPECTs.

Reply:
Yes, _STREQ is the way for strings, but exact full matches on strings are brittle.  What if later you rephrase the message or change parens to square brackets, yada yada.  I recommend trying to find the most stable parts and probe those.